gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/dbViews/views.json

```diff
@@ -12,37 +12,43 @@
+            "findWebPrefsConsumerAuthorizationByAuthCode": {
+                "map": "function(doc) {if (doc.type === 'authCode') {emit(doc.code, {'_id': doc.authorizationId})}}"
             },
-            "findAuthByAccessToken": {
-                "map": "function(doc) {if (doc.type === 'authDecision' && doc.revoked === false) {emit(doc.accessToken, {'_id': doc.clientId, 'gpiiToken': doc.gpiiToken, 'selectedPreferences': doc.selectedPreferences})}}"
+            "findOnboardedSolutionAuthorization": {
+                "map": "function(doc) {if (doc.type === 'onboardedSolutionAuthorization' && doc.revoked === false) {emit([doc.gpiiToken, doc.clientId], doc)}}"
             },
-            "findAuthDecisionByGpiiTokenAndClientId": {
-                "map": "function(doc) {if (doc.type === 'authDecision' && doc.revoked === false) {emit([doc.gpiiToken, doc.clientId], doc)}}"
+            "findUserAuthorizedAuthorizationById": {
+                "map": "function(doc) {if (doc.type === 'onboardedSolutionAuthorization' || doc.type === 'webPrefsConsumerAuthorization') emit(doc._id, doc)}"
             },
-            "findClientCredentialsTokenByClientId": {
-                "map": "function(doc) {if (doc.type === 'clientCredentialsToken' && doc.revoked === false) emit(doc.clientId, doc)}"
+            "findUserAuthorizedClientsByGpiiToken": {
+                "map": "function(doc) {if ((doc.type === 'onboardedSolutionAuthorization' || doc.type === 'webPrefsConsumerAuthorization') && doc.revoked === false) {emit(doc.gpiiToken, {'_id': doc.clientId, 'selectedPreferences': doc.selectedPreferences})}}"
             },
-            "findClientCredentialsTokenByAccessToken": {
-                "map": "function(doc) {if (doc.type === 'clientCredentialsToken' && doc.revoked === false) emit(doc.accessToken, doc)}"
+            "findPrivilegedPrefsCreatorAuthorizationById": {
+                "map": "function(doc) {if (doc.type === 'privilegedPrefsCreatorAuthorization') emit(doc._id, doc)}"
             },
-            "findAuthByClientCredentialsAccessToken": {
-                "map": "function(doc) {if (doc.type === 'clientCredentialsToken' && doc.revoked === false) {emit(doc.accessToken, {'_id': doc.clientId, 'allowAddPrefs': doc.allowAddPrefs});}}"
+            "findPrivilegedPrefsCreatorAuthorizationByClientId": {
+                "map": "function(doc) {if (doc.type === 'privilegedPrefsCreatorAuthorization' && doc.revoked === false) emit(doc.clientId, doc)}"
+            },
+            "findAuthorizationByAccessToken": {
+                "map": "function(doc) {if ((doc.type === 'onboardedSolutionAuthorization' || doc.type === 'privilegedPrefsCreatorAuthorization' || doc.type === 'webPrefsConsumerAuthorization') && doc.revoked === false) {emit(doc.accessToken, {'_id': doc.clientId, 'authorization': doc})}}"
```

Comment:
I don't think that 'onboardedSolutionAuthorization' should be here as it doesn't have an access token.